hackathon/mdakits/projects.md

```diff
@@ -0,0 +1,275 @@
+Secondary structure is a fundamental part of protein conformation. 
+Algorithms such as STRIDE [[Frishman 1995]](#Frishman1995) and 
+DSSP [[Kabsch 1983]](#Kabsch1983) calculate backbone dihedral angles
+and hydrogen-bonding energy from atomic coordinate to assign secondary 
+stucture based on a database of assignments.
+
+MDAnalysis does not currently feature a tool for the identification
+of secondary structure features (see issues 
+[#1612](https://github.com/MDAnalysis/mdanalysis/issues/1612) and 
+[#2608](https://github.com/MDAnalysis/mdanalysis/issues/2608)). 
+The ability to make atomselections based on the calculated
+secondary structure could also be of great use.
+
+### Where to start
+- Become familiar with the algorithms below and investigate
+external tools that already implement these (e.g.
+[mkdspp](https://www.mankier.com/1/mkdssp#Synopsis),
+- Add a `SecondaryStructure` class, using `AnalysisBase`, that uses
+DSSP and/or STRIDE, implemented natively or using an external
+tool as a dependency, to calculate the secondary structure of a
+selection in each frame.
+- Consider allowing the calculated secondary structure to be read
+back alongside the trajectory e.g. using the [Auxilliary
+module](https://docs.mdanalysis.org/stable/documentation_pages/auxiliary/init.html),
+and allow for atomselections based on secondary structure.
+
+
+
+
+### References
+1. <a id="Frishman1995"/>Frishman,D & Argos,P. (1995) Knowledge-based secondary structure assignment. Proteins: structure, function and genetics, 23, 566-579.
+2. <a id="Kabsch1983"/>Kabsch, W. & Sander,C. (1983) Dictionary of protein secondary structure: pattern recognition of hydrogen-bonded and geometrical features. Biopolymers, 22: 2577-2637.
+
+
+## Voronoi tessellation ##
+
+A [Voronoi diagram](https://en.wikipedia.org/wiki/Voronoi_diagram) is 
+the division of a space into regions around a set of seed points such 
+that each point in a given region is closer to that seed point than 
+any other. Voronoi tessellations can be used in various density and 
+spatial analysis, e.g. for characterizing membranes or interaction 
+surfaces/binding sites. 
+
+Voronoi tessellation and related algorithms (e.g. Delaunay triangulation) 
+have been implemented in scipy (https://docs.scipy.org/doc/scipy/reference/spatial.html#delaunay-triangulation-convex-hulls-and-voronoi-diagrams).
+
+
+## Ring puckering ##
+
+Ring groups can adopt a variety of “puckered” geometries that influence 
+their physical and chemical properties. Several formulations exist for 
+quantifying this puckering, e.g. Pickett angles [[Strauss 1970]](#Strauss1970), 
+Cremer-Pople [[Cremer 1975]](#Cremer1975), and Hill-Reilly [[Hill 2007]](#Hill2007).
+
+MDAnalysis currently features tools for assessing puckering in 5-membered 
+rings in our [nucleic acid analysis module](https://docs.mdanalysis.org/stable/documentation_pages/analysis/nuclinfo.html); 
+however, such capabilities are lacking for 6 (or higher) membered rings.
+
```

Comment:
Do we need a where to start here?

1. identify rings
2. calculate CP angles
3. calculate HR angles
4. test against known test cases

Reply:
I have no idea, that's not something I've ever done. I think you originally proposed the project. We could merge "as is" and then you can update later to make it sound the way that you imagined.

Reply:
I think what I'm pointing out is an inconsistency across the summaries, we have some with objectives, some with where to start,

Reply:
If someone wants to make it homogenous then that's fine. I think we should use whatever works. I tend to use "Objectives" as a clear "This is what's needed and when you've done it then you have achieved something" but that might not always be the best way to phrase an open-ended idea.